presto-druid/src/main/java/com/facebook/presto/druid/DruidAggregationProjectConverter.java

```diff
@@ -0,0 +1,187 @@
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.spi.type.TypeManager;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.Slice;
+import org.joda.time.DateTimeZone;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.druid.DruidErrorCode.DRUID_PUSHDOWN_UNSUPPORTED_EXPRESSION;
+import static com.facebook.presto.druid.DruidExpression.derived;
+import static com.facebook.presto.druid.DruidPushdownUtils.getLiteralAsString;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class DruidAggregationProjectConverter
+        extends DruidProjectExpressionConverter
+{
+    private static final Map<String, String> PRESTO_TO_DRUID_OPERATORS = ImmutableMap.of(
+            "-", "SUB",
+            "+", "ADD",
+            "*", "MULT",
+            "/", "DIV");
+    private static final String FROM_UNIXTIME = "from_unixtime";
+
+    private final FunctionMetadataManager functionMetadataManager;
+    private final ConnectorSession session;
+
+    public DruidAggregationProjectConverter(TypeManager typeManager, FunctionMetadataManager functionMetadataManager, StandardFunctionResolution standardFunctionResolution, ConnectorSession session)
+    {
+        super(typeManager, standardFunctionResolution);
+        this.functionMetadataManager = requireNonNull(functionMetadataManager, "functionMetadataManager is null");
+        this.session = requireNonNull(session, "session is null");
+    }
+
+    @Override
+    public DruidExpression visitCall(
+            CallExpression call,
+            Map<VariableReferenceExpression, DruidQueryGeneratorContext.Selection> context)
+    {
+        Optional<DruidExpression> basicCallHandlingResult = basicCallHandling(call, context);
+        if (basicCallHandlingResult.isPresent()) {
+            return basicCallHandlingResult.get();
+        }
+        FunctionMetadata functionMetadata = functionMetadataManager.getFunctionMetadata(call.getFunctionHandle());
+        Optional<OperatorType> operatorTypeOptional = functionMetadata.getOperatorType();
+        if (operatorTypeOptional.isPresent()) {
+            OperatorType operatorType = operatorTypeOptional.get();
+            if (operatorType.isArithmeticOperator()) {
+                return handleArithmeticExpression(call, operatorType, context);
+            }
+            if (operatorType.isComparisonOperator()) {
+                throw new PrestoException(DRUID_PUSHDOWN_UNSUPPORTED_EXPRESSION, "Unsupported operator: " + call + " to pushdown for Druid connector ");
+            }
+        }
+        return handleFunction(call, context);
+    }
+
+    @Override
+    public DruidExpression visitConstant(
+            ConstantExpression literal,
+            Map<VariableReferenceExpression, DruidQueryGeneratorContext.Selection> context)
```

Comment:
one param per line; same for all other functions in this class.

Reply:
yep